### PR TITLE
Feature/tests fixes demos

### PR DESCRIPTION
**Pull request by ivan-pavlov**

### CODE REVIEW

**Review by eddelbuettel — APPROVED** [review]
looks good at a first glance

maybe add a test for LDA ?

**Comment by eddelbuettel** [comment]
Poking @coatless ...  Any comment or concerns?

File: R/functions.R [hunk 1]
```diff
@@ -666,24 +671,36 @@ df2vw <- function(data, file_path, namespaces = NULL, keep_space = NULL,
...
+        }
         ret
     }
 
     # parse categorical variables
     parsingVar <- function(x, keepSpace) {
-        # remove leading and trailing spaces, then remove special characters
-        # then remove isolated underscores.
-        if (!keepSpace)
+        if (!keepSpace) # Replace special characters
```
Comment (coatless) [inline R/functions.R]: Could you emphasize the code blocks here since you've moved the comments down.

File: R/functions.R [hunk 2]
```diff
@@ -666,24 +671,36 @@ df2vw <- function(data, file_path, namespaces = NULL, keep_space = NULL,
...
             spch <- specCharSpace
-        else
+        else # Replace special characters, but keep spaces
             spch <- specChar
         gsub(spch, '_', x)
     }
-
+    # parsingVar <- function(x, keepSpace, fixed) {
```
Comment (coatless) [inline R/functions.R]: Any reason for adding in commented out code?

File: R/functions.R [hunk 3]
```diff
@@ -815,41 +832,55 @@ df2vw <- function(data, file_path, namespaces = NULL, keep_space = NULL,
...
+        # Enable special parsing for features stated in "keep_space" and "fixed" arguments
+        Header[[nsN]][!Index[[nsN]]] <- ifelse(
+            test = Header[[nsN]][!Index[[nsN]]] %in% fixed,
+            yes = Header[[nsN]][!Index[[nsN]]],
+            no = paste0("eval(parse(text = 'parsingVar(",
+                        Header[[nsN]][!Index[[nsN]]],
+                        ", keepSpace = ", Header[[nsN]][!Index[[nsN]]] %in% keep_space, ")'))")
+        )
+        
+        # Header[[nsN]][!Index[[nsN]]] <- paste0("eval(parse(text = 'parsingVar(",
```
Comment (coatless) [inline R/functions.R]: Ditto on commented code. This is where code diffs are useful.

File: R/functions.R [hunk 4]
```diff
@@ -815,41 +832,55 @@ df2vw <- function(data, file_path, namespaces = NULL, keep_space = NULL,
...
-        formatDataVW <- paste0(formatDataVW, collapse = '|')
-    } else {
-        formatDataVW_label <- formatDataVW[1]
-        formatDataVW <- paste0(formatDataVW[2:length(formatDataVW)], collapse = '|')
-        formatDataVW <- paste0(formatDataVW_label, ' |', formatDataVW)
-    }
-
+    formatDataVW <- paste0(formatDataVW, collapse = " |")
+    
+    # Old version, when "'tag" wasn't used
```
Comment (coatless) [inline R/functions.R]: Ditto on commented code. This is where code diffs are useful.

File: demo/00Index [hunk 5]
```diff
@@ -1,2 +1,3 @@
 example_1		Basic package usage
 example_2       Using data.frame as input data
+example_3       LDA example
```
Comment (coatless) [inline demo/00Index]: Perhaps `pkg_overview`, `df_example`, `perform_lda`?

**Review by coatless — CHANGES_REQUESTED** [review]
Perhaps switch the example file names from `example_1` .. `example_3` to the underlying motivation?

File: R/functions.R (hunk 2 above)
Comment (eddelbuettel) [reply R/functions.R]: Ah well I do that too all the time.

File: demo/00Index (hunk 5 above)
Comment (eddelbuettel) [reply demo/00Index]: Not a bad idea.  Maube `rvw_overview`, `rvw_df_examples`, `rvw_lda_example` ?   `demo()` lists all, even though it is chunked it is nice if they 'look' similar.  My $0.02.

File: R/functions.R (hunk 2 above)
Comment (ivan-pavlov) [reply R/functions.R]: I thought I would return to this variant, but there's actually no point keeping it anymore.
So I will just delete all these lines.

File: demo/00Index (hunk 5 above)
Comment (ivan-pavlov) [reply demo/00Index]: yes, sounds really reasonable

**Comment by eddelbuettel** [comment]
Looks good to me.

**Review by coatless — APPROVED** [review]

**Pull request merged**